cinema/views.py

```diff
@@ -31,8 +38,21 @@ class CinemaHallViewSet(viewsets.ModelViewSet):
 
 
 class MovieViewSet(viewsets.ModelViewSet):
-    queryset = Movie.objects.all()
-    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        queryset = Movie.objects.all()
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if actors:
+            queryset = queryset.filter(actors__id__icontains=actors)
```

Comment:
Filtering by `actors__id__icontains` is not appropriate for an ID field. Consider using `actors__id=actors` if `actors` is expected to be a single ID, or adjust the logic to handle multiple IDs if necessary.